modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -752,6 +763,16 @@ void generateApis(List<File> files, List<OperationsMap> allOperations, List<Mode
 
     }
 
+    // checking if apiController file is already existed for spring generator
+    private boolean apiFilePreCheck(String filename, String generator, String templateName, String apiControllerTemplate){
+        File apiFile = new File(filename);
+        if(apiFile.exists() && config.getName().equals(generator) && templateName.equals(apiControllerTemplate)){
```

Comment:
@MelleD , have removed else and directly returned Boolean.

Reply:
Better :-) 
But you can also remove completely the whole if, right?

`return apiFile.exists() || config.getName().equals(generator) ||templateName.equals(apiControllerTemplate)`

Reply:
below logic I am applying here:
if file is already existed && generator is spring && template is controller, then only skip creating the file , other wise we have to create files if any of the conditions holds false.
but in above suggestion it will create file if any of the condition holds true.
If I remove if I will have to add as below:
return apiFile.exists() && config.getName().equals(generator) && templateName.equals(apiControllerTemplate)

additionally !apiFilePreCheck where I m using this function.
Please let me know if it is needed ?

Reply:
Yes or set brackets !(condition).
Up to you but to make a if and directly return a Boolean looks redundant :-)

Reply:
Pushed the suggested changes 👍